pkg/util/strategicpatch/patch.go

```diff
@@ -167,6 +213,12 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	if t.Kind() != reflect.Map {
 		// Maybe in the future add a "concat" mode that doesn't
 		// uniqify.
+		if !reflect.DeepEqual(original, patch) {
+			dw.WriteRemoval(fmt.Sprintf("%v", original))
+			dw.WriteAddition(fmt.Sprintf("%v", patch))
+		} else {
+			dw.WriteSame(fmt.Sprintf("%v", original))
+		}
 		both := append(original, patch...)
 		return uniqifyScalars(both), nil
 	}
```

Comment:
@kbeecher, IIUC, what these two lines do is finding the elements in `patch` that doesn't belong to `original`, and adding these elements to `original`.